api/apps/geoprocessing/src/modules/planning-area/planning-units-grid/planning-area-grid-tiles.service.ts

```diff
@@ -16,29 +16,25 @@ export class PlanningAreaGridTilesService {
     private readonly tileService: TileService,
   ) {}
 
-  buildCustomPlanningAreaWhereQuery(planningAreaId: string): string {
-    /**
-     * @todo this generation query is a bit...
-     */
-    let whereQuery = `project_id = '${planningAreaId}'`;
-
-    return whereQuery;
-  }
-
   /**
    * @todo get attributes from Entity, based on user selection
    */
   public findTile(tileSpecification: TileSpecification): Promise<Buffer> {
     const { z, x, y, planningAreaId } = tileSpecification;
     const attributes = 'project_id as planningAreaId';
-    const table = this.repository.metadata.tableName;
-    const customQuery = this.buildCustomPlanningAreaWhereQuery(planningAreaId);
+
+    const qb = this.repository
+      .createQueryBuilder('pug')
+      .select('ppu.project_id', 'project_id')
+      .addSelect('pug.the_geom', 'the_geom')
+      .innerJoin('projects_pu', 'ppu', 'pug.id = ppu.geom_id')
+      .where(`ppu.project_id = '${planningAreaId}'`);
```

Comment:
We tried to use `where()` params here, but a problem arose. Since generated sql is passed as an argument to another function (`getTile` of `TileService`), which internally creates another query builder, params get overwritten and the query doesn't work as expected 🤔